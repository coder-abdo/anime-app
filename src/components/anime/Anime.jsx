import React from "react";
import { Link } from "react-router-dom";
import { AnimeContainer } from "./AnimeStyle";
export const Anime = ({ image, title, url, score, id }) => (
  <AnimeContainer>
    <div className="more-detail">
      <Link data-testid="more-details" className="more-link" to={`/${id}`}>
        more details
      </Link>
    </div>
    <div className="imgContainer">
      <img src={image} alt={title} />
    </div>
    <div className="details">
      <h3 data-testid="anime-title">{title}</h3>
      <footer>
        <h4 data-testid="anime-score">score: {score}</h4>
        <a href={url}>myAnime list page</a>
      </footer>
    </div>
  </AnimeContainer>
);
