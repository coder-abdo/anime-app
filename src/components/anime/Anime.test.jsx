import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import { Router } from "react-router";
import { createBrowserHistory } from "history";
import "@testing-library/jest-dom";
import { Anime } from "./Anime";
afterEach(cleanup);
describe("render Anime ", () => {
  const mockProps = {
    image: "some image url",
    title: "anime title",
    url: "dummy url",
    id: 12,
    score: 2
  };
  const history = createBrowserHistory();
  const { container, getByTestId } = render(
    <Router history={history}>
      <Anime {...mockProps} />
    </Router>
  );
  const animeTitle = getByTestId("anime-title");
  const animeScore = getByTestId("anime-score");
  test("render Anime Component with props properly", () => {
    expect(container).toBeInTheDocument();
  });
  test("more details after clicking the more detail button", () => {
    const { getByTestId } = render(
      <Router history={history}>
        <Anime {...mockProps} />
      </Router>
    );
    expect(history.length).toBe(1);
    fireEvent.click(getByTestId("more-details"));
    expect(history.length).toBe(2);
  });
  test("render anime title to be anime title", () => {
    expect(animeTitle.textContent).toBe("anime title");
  });
  test("render anime title to be 2", () => {
    expect(animeScore.textContent).toBe("score: 2");
  });
});
