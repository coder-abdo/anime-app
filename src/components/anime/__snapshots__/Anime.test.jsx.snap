// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Anime  render Anime Component with props properly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm gItdEj"
  >
    <div
      class="more-detail"
    >
      <a
        class="more-link"
        data-testid="more-details"
        href="/12"
      >
        more details
      </a>
    </div>
    <div
      class="imgContainer"
    >
      <img
        alt="anime title"
        src="some image url"
      />
    </div>
    <div
      class="details"
    >
      <h3
        data-testid="anime-title"
      >
        anime title
      </h3>
      <footer>
        <h4
          data-testid="anime-score"
        >
          score: 2
        </h4>
        <a
          href="dummy url"
        >
          myAnime list page
        </a>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
